src/test_l3_dsFPD.c

```diff
@@ -0,0 +1,691 @@
+    {"dsFPD_LED_DEVICE_NONE", (int32_t)dsFPD_LED_DEVICE_NONE},
+    {"dsFPD_LED_DEVICE_ACTIVE", (int32_t)dsFPD_LED_DEVICE_ACTIVE},
+    {"dsFPD_LED_DEVICE_STANDBY", (int32_t)dsFPD_LED_DEVICE_STANDBY},
+    {"dsFPD_LED_DEVICE_WPS_CONNECTING", (int32_t)dsFPD_LED_DEVICE_WPS_CONNECTING},
+    {"dsFPD_LED_DEVICE_WPS_CONNECTED", (int32_t)dsFPD_LED_DEVICE_WPS_CONNECTED},
+    {"dsFPD_LED_DEVICE_WPS_ERROR", (int32_t)dsFPD_LED_DEVICE_WPS_ERROR},
+    {"dsFPD_LED_DEVICE_FACTORY_RESET", (int32_t)dsFPD_LED_DEVICE_FACTORY_RESET},
+    {"dsFPD_LED_DEVICE_USB_UPGRADE", (int32_t)dsFPD_LED_DEVICE_USB_UPGRADE},
+    {"dsFPD_LED_DEVICE_SOFTWARE_DOWNLOAD_ERROR", (int32_t)dsFPD_LED_DEVICE_SOFTWARE_DOWNLOAD_ERROR},
+    {"dsFPD_LED_DEVICE_MAX", (int32_t)dsFPD_LED_DEVICE_MAX},
+    {NULL, -1}
+};
+
+static void readAndDiscardRestOfLine(FILE* in)
+{
+    int c;
+    while ( (c = fgetc(in)) != EOF && c != '\n');
+}
+
+
+static int selectIndicator()
+{
+
+    UT_LOG_MENU_INFO(" \t  Supported Indicators are:");
+    UT_LOG_MENU_INFO("------------------------------------------");
+    for (int indicator = 0; indicator <= dsFPD_INDICATOR_MAX; indicator++)
+    {
+        UT_LOG_INFO("\t%d.  %-20s\n", indicator, UT_Control_GetMapString(dsFrontPanelIndicatorTable, indicator));
+    }
+    UT_LOG_MENU_INFO("----------------------------------------------------------");
+    UT_LOG_MENU_INFO("Select Indicator : ");
+    scanf("%d", &eIndicator);
+    readAndDiscardRestOfLine(stdin);
+    if(eIndicator < dsFPD_INDICATOR_MESSAGE || eIndicator >=dsFPD_INDICATOR_MAX )
+    {
+        UT_LOG_INFO("Unsupported Indicator %d",eIndicator);
+        return 1;
+    }
+    return 0;
+}
+
+/**
+* @brief Initialization of the Device settings - Front Panel Module
+*
+* This test provides a scope to Intialize the Front Panel module.
+
+* **Pre-Conditions:** None@n
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 1 to before running any test.
+*
+*/
+
+void test_l3_dsFPD_hal_Init(void)
+{
+   gTestID = 1;
+   dsError_t status = dsERR_NONE;
+
+   UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+   // Step 1: Call dsFPInit()
+   UT_LOG_INFO("Calling dsFPInit()");
+   status = dsFPInit();
+   UT_LOG_INFO("Result dsFPInit() dsError_t:[%s]",UT_Control_GetMapString(dsFrontPanelErrorCodeTable, status));
+   assert(status == dsERR_NONE);
+
+   UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+/**
+* @brief This test provides a scope to set Front panel LED state.
+*
+* This test case provides a scope to  set the State of FP indicator to ON and OFF.
+*
+* **Pre-Conditions:** @n
+* Front panel Module should be intialized through Test 1 before calling this test.
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 2 and provide indicator index and State to be set.
+*
+*/
+void test_l3_dsFPD_hal_SetFPState(void)
+{
+    gTestID = 2;
+    int ret = 0;
```

Comment:
use int32_t instead of int . correct all places

Reply:
updated